internal/cloud/gke.go

```diff
@@ -629,21 +646,24 @@ func tpuMachineType(accel string, tpuRequest int) (string, error) {
 	return "", fmt.Errorf("invalid accelerator: %v", accel)
 }
 
-func waitForGkeOp(svc *containerv1beta1.Service, c GKEContext, operation *containerv1beta1.Operation) error {
-	operationWaitTimeout := 30 * time.Minute
+func waitForGkeOp(ctx context.Context, svc *containerv1beta1.Service, c GKEContext, operation *containerv1beta1.Operation) error {
 	operationPollInterval := 5 * time.Second
 
-	for start := time.Now(); time.Since(start) < operationWaitTimeout; time.Sleep(operationPollInterval) {
-		if op, err := svc.Projects.Locations.Operations.Get(c.OpName(operation.Name)).Do(); err == nil {
+	for {
+		select {
+		case <-ctx.Done():
```

Comment:
Why is this logic changed?

Reply:
PS: I think you dont need to check the context Done() here, I think you should just pass the context into the Get() call using a different Get method that allows for contexts